src/crud_ops.rs

```diff
@@ -143,6 +225,51 @@ impl Client {
         counter!(metrics::total_delete_ops).increment(1);
         with_retries!(self, self.delete_impl(path).await)
     }
+
+    async fn bulk_delete_impl(&self, paths: &Vec<Path>) -> crate::Result<Vec<(Path, crate::Error)>> {
+        let stream = stream::iter(paths.iter().map(|path| Ok(path.clone()))).boxed();
+        let indexed_results = self.store.delete_stream(stream)
+            .enumerate()
+            .filter_map(|(index, result)| async move {
+                Some((index, result))
+            }).collect::<Vec<(usize, Result<Path, object_store::Error>)>>().await;
```

Comment:
My suggestion was confusing, you can skip the `enumerate` and `filter_map` here entirely and just `collect` like:
```suggestion
        let results = self.store.delete_stream(stream).collect::<Vec<_>>().await;
```
Then you add enumerate to the `.into_iter()` below. That should be cleaner.